lldb/source/Plugins/OperatingSystem/SwiftTasks/OperatingSystemSwiftTasks.cpp

```diff
@@ -105,7 +111,12 @@ bool OperatingSystemSwiftTasks::UpdateThreadList(ThreadList &old_thread_list,
   LLDB_LOG(log, "OperatingSystemSwiftTasks: Updating thread list");
 
   for (const ThreadSP &real_thread : core_thread_list.Threads()) {
-    std::optional<uint64_t> task_id = FindTaskId(*real_thread);
+    std::optional<uint64_t> task_id;
+    std::optional<std::string> task_name;
+    if (auto task_addr = FindTaskAddress(*real_thread)) {
+      task_id = FindTaskId(*task_addr);
+      task_name = FindTaskName(*task_addr, m_process);
+    }
```

Comment:
The control flow becomes indirect with this change: below we check for `task_id`, which is only set if `task_addr` also was ok; it's the kind of thing that adds cognitive burden when reading code. IMO we should also avoid the declare-then-initialize pattern, which is what causes this indirectness of the control flow.

Here's a suggestion: flip the condition below to `!task_addr`, early return then.

Unconditionally initialize task_id/task_name after the early returns and have another early return if we fail to obtain the task_id, as those are two separate errors we may wish to log.

Reply:
check out the updates (including handling of the two separate errors)